x/stake/handler_sidechain.go

```diff
@@ -136,6 +140,135 @@ func handleMsgEditSideChainValidator(ctx sdk.Context, msg MsgEditSideChainValida
 	}
 }
 
+func handleMsgCreateSideChainValidatorWithVoteAddr(ctx sdk.Context, msg MsgCreateSideChainValidatorWithVoteAddr, k keeper.Keeper) sdk.Result {
+	if scCtx, err := k.ScKeeper.PrepareCtxForSideChain(ctx, msg.SideChainId); err != nil {
+		return ErrInvalidSideChainId(k.Codespace()).Result()
+	} else {
+		ctx = scCtx
+	}
+
+	// check to see if BEP126 has upgraded
+	if !sdk.IsUpgrade(sdk.BEP126) {
+		return types.ErrBadValidatorSideVoteAddr(k.Codespace()).Result()
+	}
+
+	// check to see if the pubkey or sender has been registered before
+	_, found := k.GetValidator(ctx, msg.ValidatorAddr)
+	if found {
+		return ErrValidatorOwnerExists(k.Codespace()).Result()
+	}
+
+	_, found = k.GetValidatorBySideConsAddr(ctx, msg.SideConsAddr)
+	if found {
+		return ErrValidatorSideConsAddrExist(k.Codespace()).Result()
+	}
+
+	_, found = k.GetValidatorBySideVoteAddr(ctx, msg.SideVoteAddr)
+	if found {
+		return ErrValidatorSideVoteAddrExist(k.Codespace()).Result()
+	}
+
+	minSelfDelegation := k.MinSelfDelegation(ctx)
+	if msg.Delegation.Amount < minSelfDelegation {
+		return ErrBadDelegationAmount(DefaultCodespace,
+			fmt.Sprintf("self delegation must not be less than %d", minSelfDelegation)).Result()
+	}
+	if msg.Delegation.Denom != k.GetParams(ctx).BondDenom {
+		return ErrBadDenom(k.Codespace()).Result()
+	}
+
+	// self-delegate address will be used to collect fees.
+	feeAddr := msg.DelegatorAddr
+	validator := NewSideChainValidator(feeAddr, msg.ValidatorAddr, msg.Description, msg.SideChainId, msg.SideConsAddr, msg.SideFeeAddr, msg.SideVoteAddr)
+	commission := NewCommissionWithTime(
+		msg.Commission.Rate, msg.Commission.MaxRate,
+		msg.Commission.MaxChangeRate, ctx.BlockHeader().Time,
+	)
+	var err sdk.Error
+	validator, err = validator.SetInitialCommission(commission)
+	if err != nil {
+		return err.Result()
+	}
+
+	k.SetValidator(ctx, validator)
+	k.SetValidatorByConsAddr(ctx, validator) // here consAddr is the sideConsAddr
+	k.SetValidatorBySideVoteAddr(ctx, validator)
+	k.SetNewValidatorByPowerIndex(ctx, validator)
+	k.OnValidatorCreated(ctx, validator.OperatorAddr)
+
+	// move coins from the msg.Address account to a (self-delegation) delegator account
+	// the validator account and global shares are updated within here
+	_, err = k.Delegate(ctx, msg.DelegatorAddr, msg.Delegation, validator, true)
+	if err != nil {
+		return err.Result()
+	}
+
+	return sdk.Result{
+		Tags: sdk.NewTags(
+			tags.DstValidator, []byte(msg.ValidatorAddr.String()),
+			tags.Moniker, []byte(msg.Description.Moniker),
+			tags.Identity, []byte(msg.Description.Identity),
+		),
+	}
+}
+
+func handleMsgEditSideChainValidatorWithVoteAddr(ctx sdk.Context, msg MsgEditSideChainValidatorWithVoteAddr, k keeper.Keeper) sdk.Result {
+	if scCtx, err := k.ScKeeper.PrepareCtxForSideChain(ctx, msg.SideChainId); err != nil {
```

Comment:
Please refer to this PR: https://github.com/bnb-chain/bnc-cosmos-sdk/pull/318
There is some change on how we support edit side chain validator